### ISSUE TITLE
Is the output of YouTubeDNN the sample-softmax-loss? 

### ISSUE THREAD

**Issue by bing-zhao** [shenweichen/DeepMatch#39]
**Describe the question(问题描述)**
Would any one please help to clarify my understanding the YouTubeDNN and its usage with negative sampling? 

In the example "[YoutubeDNN/MIND with sampled softmax](https://deepmatch.readthedocs.io/en/latest/Examples.html#youtubednn-mind-with-sampled-softmax)", the model architecture is created using the following line, 

```
model = YoutubeDNN(user_feature_columns, item_feature_columns, num_sampled=5, user_dnn_hidden_units=(64, embedding_dim))
model.compile(optimizer="adam", loss=sampledsoftmaxloss)
```

By checking the [source code](https://github.com/shenweichen/DeepMatch/blob/ea69d9cc9aec3742ae70f487b3ec9491fdfb14a8/deepmatch/models/youtubednn.py#L68), 

`output = SampledSoftmaxLayer(num_sampled=num_sampled)(
        [pooling_item_embedding_weight, user_dnn_out, item_features[item_feature_name]])`

In [SampledSoftmanxLayer](https://github.com/shenweichen/DeepMatch/blob/ea69d9cc9aec3742ae70f487b3ec9491fdfb14a8/deepmatch/layers/core.py#L45), during the calculation of sampled softmax loss, the above three arguments corresponds to 
* `weights=pooling_item_embedding_weight`, i.e., the movie id embedding matrix
* `labels=item_features[item_feature_name]`, i.e., the movie ids
* `inputs=user_dnn_out`, i.e., the user embedding

**Question 1.** Am I right to say that the final output from the YouTubeDNN model is the loss value (i.e., 1-dimension)? 

If it is yes for the above question, I am confused with the following line
`history = model.fit(train_model_input, train_label,  batch_size=256, epochs=1, verbose=1, validation_split=0.0, )`

Because in the example, `train_label` is defined as a binary variable containing 0 or 1, indicating whether the instance is a positive or negative sample. In addition, since `negsample = 0`, the values in `train_label` are all `1s` (i.e., no negative samples). Thus, 

**Question 2.** How is `train_label` being used during the training when calculating the loss? 

**Question 3.** If I set `negsample = 1` or any non-zero values, how am I supposed to use YouTubeDNN method? 

Thanks. Any advice is appreciated. 

**Operating environment(运行环境):**
 - python version [e.g. 3.7]
 - tensorflow version [e.g. 2.3.0,]
 - deepmatch version [e.g. 0.1.3,]

**Comment by shenweichen** [comment]
1. yes the final output of the YouTubeDNN model is the loss value.In this example,we do not need to feed the negtive sample to the model because we use the `sampledsoftmaxlayer`, it will sample negtive samples during training phase.
2. the positive label and the sampled negative labels will be used to calculate the softmax loss
3. if you set nesample=1,you need to modify to calculate the loss with positive and negative labels you feed to the model

**Comment by bing-zhao** [comment]
Thank you for your reply!
